ipatests/test_integration/test_epn.py

```diff
@@ -494,3 +497,126 @@ def test_EPN_delay_config(self, cleanupmail):
         self.master.put_file_contents('/etc/ipa/epn.conf', epn_conf)
         result = tasks.ipa_epn(self.master, raiseonerr=False)
         assert "smtp_delay cannot be less than zero" in result.stderr_text
+
+    def generate_user_batch(self):
+        """
+        Prepare large (10k) batch of users with various attributes, namely
+        without passwords, gradually expiring passwords, unicode names and
+        unicode e-mails
+        :return: None
+        """
+        USERS_IN_GROUP = 2000
+        users = deque()  # entries are stored as tuple (username, expiration)
+
+        # Users without passwords
+        for i in range(USERS_IN_GROUP):
+            username = "user_wo_pass_%d" % i
+            tasks.user_add(self.master, username)
+            users.append((username, -1))
+
+        # Users with passwords gradually expiring
+        for i in range(USERS_IN_GROUP):
+            username = "user_with_pass_%d" % i
+            # TODO: Compute
+            krbpasswordexpiration = datetime_to_generalized_time(
+                datetime.datetime.utcnow() + datetime.timedelta(days=7)
```

Comment:
Assuming this runs really really fast the date is going to very only slightly. Would it be better to move this out of the loop and save all the sys calls to get time? I don't think a few seconds will make a difference in the test.